src/m365/entra/commands/group/group-set.spec.ts

```diff
@@ -0,0 +1,367 @@
+    assert.deepStrictEqual(patchRequestStub.lastCall.args[0].data, updateGroupRequest);
+  });
+
+  it('successfully updates group with owners specified by ids and removes current owners', async () => {
+    sinon.stub(request, 'get').resolves({ value: [{ id: '717f1683-00fa-488c-b68d-5d0051f6bcfa' }] });
+    sinon.stub(request, 'patch').resolves();
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'PATCH') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'DELETE') {
+        return {
+          responses: [
+            {
+              status: 204,
+              body: {}
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { id: groupId, description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Public', ownerIds: userIds.join(',') } });
+    assert.deepStrictEqual(postStub.firstCall.args[0].data.requests, addOwnersRequest);
+  });
+
+  it('successfully updates group with members specified by user names and removes current members', async () => {
+    sinon.stub(request, 'get').resolves({ value: [{ id: '717f1683-00fa-488c-b68d-5d0051f6bcfa' }] });
+    sinon.stub(entraUser, 'getUserIdsByUpns').withArgs(userUpns).resolves(userIds);
+    sinon.stub(entraGroup, 'getGroupIdByDisplayName').withArgs('Microsoft 365 Group').resolves(groupId);
+    sinon.stub(request, 'patch').resolves();
+
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'PATCH') {
+        return {
+          responses: Array(2).fill({
+            status: 204,
+            body: {}
+          })
+        };
+      }
+
+      if (opts.url === 'https://graph.microsoft.com/v1.0/$batch' &&
+        opts.data.requests[0].method === 'DELETE') {
+        return {
+          responses: [
+            {
+              status: 204,
+              body: {}
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { displayName: 'Microsoft 365 Group', description: 'Microsoft 365 group', mailNickname: 'Microsoft365Group', visibility: 'Private', memberUserNames: userUpns.join(','), verbose: true } });
+    assert.deepStrictEqual(postStub.firstCall.args[0].data.requests, addMembersRequest);
+  });
+
+  it('handles API error when adding users to a group', async () => {
+    sinon.stub(entraGroup, 'getGroupIdByDisplayName').withArgs('Microsoft 365 Group').resolves(groupId);
```

Comment:
Since we use this in quite some tests, maybe it's an idea to put it in the `before` function so it's available to all tests and we're avoiding repetition.

Reply:
ok